api/handlers/collectionshandler.py

```diff
@@ -57,6 +57,8 @@ def put(self, **kwargs):
         mongo_validator, payload_validator = self._get_validators()
 
         payload = self.request.json_body or {}
+        if not payload:
+            self.abort(400, 'Bad Request')
```

Comment:
"Bad Request" is the title of the 400 error, you can put something more descriptive here. "PUT request body cannot be empty" or something.